src/VecSim/algorithms/hnsw/hnsw_serializer.h

```diff
@@ -91,16 +99,28 @@ HNSWIndexMetaData HNSWIndex<DataType, DistType>::checkIntegrity() const {
     if (num_deleted != this->numMarkedDeleted) {
         return res;
     }
+
+    // Validate that each node's in-degree is coherent with the in-degree observed by the
+    // outgoing edges.
+    for (size_t i = 0; i < this->curElementCount; i++) {
+        for (size_t l = 0; l <= getGraphDataByInternalId(i)->toplevel; l++) {
+            LevelData &cur = this->getLevelData(i, l);
+            if (cur.totalIncomingLinks != inbound_connections_num[i][l]) {
+                return res;
+            }
```

Comment:
What's this? When can that happen?

Reply:
This is the sanity check for the new field. It should never happen, and if so, it means that the index is corrupted.